brainglobe_utils/qtpy/collapsible_widget.py

```diff
@@ -0,0 +1,162 @@
+    """
+
+    toggled_signal_with_self = Signal(QCollapsible, bool)
+
+    def __init__(
+        self,
+        title: str = "",
+        parent: Optional[QWidget] = None,
+        expanded_icon: Optional[str] = "▲",
+        collapsed_icon: Optional[str] = "▼",
+    ):
+        """
+        Initializes a new CollapsibleWidget instance.
+
+        Parameters
+        ----------
+        title : str, optional
+            The title of the CollapsibleWidget.
+        parent : QWidget or None, optional
+            The parent widget.
+        expanded_icon : str or None, optional
+            The icon for the expanded state.
+        collapsed_icon : str or None, optional
+            The icon for the collapsed state.
+        """
+        super().__init__(title, parent, expanded_icon, collapsed_icon)
+        self.currently_expanded = False
+
+        self.toggled.connect(self._on_toggle)
+
+    def _on_toggle(self, state):
+        """
+        Handles the toggled signal by emitting the custom signal
+         with the CollapsibleWidget instance and the new state.
+
+        Parameters
+        ----------
+        state : bool
+            The new state of the CollapsibleWidget
+            (True if expanded, False if collapsed).
+        """
+        self.toggled_signal_with_self.emit(self, state)
+
+
+class CollapsibleWidgetContainer(QGroupBox):
+    """
+    Container for multiple CollapsibleWidgets with the ability to add,
+    remove, and synchronize their states.
+
+    Methods
+    -------
+    add_widget(QWidget or CollapsibleWidget)
+        Adds a widget to the CollapsibleWidgetContainer.
+    remove_drawer(QWidget or CollapsibleWidget)
+        Removes a widget from the CollapsibleWidgetContainer.
+    _update_drawers(signalling_drawer, state)
+        Private method to synchronize drawer states.
+    """
+
+    def __init__(self):
+        """
+        Initializes a new CollapsibleWidgetContainer instance.
+        """
+        super().__init__()
+        self.setLayout(QVBoxLayout())
+
+        self.layout().setAlignment(Qt.AlignTop)
+        self.layout().setSpacing(0)
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.collapsible_widgets: List[CollapsibleWidget] = []
+
+    @overload
+    def add_widget(self, widget: QWidget):
+        ...
+
+    @overload
+    def add_widget(self, widget: CollapsibleWidget):
+        ...
+
+    def add_widget(self, widget: Union[QWidget, CollapsibleWidget]):
+        """
+        Adds a QWidget or a CollapsibleWidget to the chest.
+
+        Parameters
+        ----------
+        widget : QWidget or CollapsibleWidget
+            The widget instance to be added.
+        """
+        if isinstance(widget, CollapsibleWidget):
+            self.collapsible_widgets.append(widget)
+            widget.toggled_signal_with_self.connect(self._update_drawers)
+
+        self.layout().addWidget(widget, 0, Qt.AlignTop)
+
+    @overload
+    def remove_widget(self, widget: QWidget):
```

Comment:
OK, I agree - but why are both cases (removing a collapsible, removing a non-collapsible) not sufficiently covered by `remove_widget(self, widget: Union[QWidget, CollapsibleWidget])`? Would they be calling things in the superclass `QCollapsible` otherwise?

Reply:
I see what you mean. I don't think so, seems I was using the `@overload` decorator inappropriately. I'll remove the overloaded functions and just leave the implementations.